syntax = "proto3";

service CreateWGConnection {
    // creates the initial connection
    rpc StartConnection (ConnectionInit) returns (ConnectionResp) {}
}

message ConnectionInit {
    string email = 1;
    string clientPubKey = 2;
    string accessToken = 3;
    int32 deviceId = 4;
}


message ConnectionResp {
    string email = 1;
    string serverPubKey = 2;
    int32  portNumber = 3;
    string serverIPAddr = 4;
    bytes certificateFileCrt = 5;
}
